cfblog/middleware.py

```diff
@@ -0,0 +1,38 @@
+__author__ = 'vinay'
+from django.conf import settings
+from django.http.response import HttpResponse
+
+from .models import CmsPage
+from .settings import CMS_AUTH_TEST_FUNC
+from .utils import parse_cms_template
+from .views import cms_page_index
+
+
+class CmsPageMiddleware(object):
+
+    def process_response(self, request, response):
+        # ignore all ajax, static and media requests
+        if request.is_ajax() or \
+                request.path.startswith(settings.STATIC_URL or '///') or \
+                request.path.startswith(settings.MEDIA_URL or '///'):
+            return response
+
+        # we return the original response
+        # if the response is given by cms_page_index by catching url pattern
+        if not request.resolver_match or request.resolver_match.func is not cms_page_index:
```

Comment:
Do the opposite of this condition and return response.

Basically like above, where you checked for is_ajax() etc, you should always look for "bad conditions" and return, and then start processing, else you have meaningless nesting. Think of everything that can go wrong, and check and return, and then start processing.